### PR TITLE
cclient/serializer/json: get_transactions_to_approve serialization

### LINKED ISSUE #874: cclient/serializer/json:get_transactions_to_approve - serialize response & deserialize request
This needs implementation, touches:

`cclient/serializer/serializer.h`
`cclient/serializer/json/json_serializer.c`
`cclient/serializer/json/get_transactions_to_approve.{c,h}`
`cclient/serializer/json/tests/get_transactions_to_approve.c`

### PR DESCRIPTION
**Pull request by howjmay**
cclient/serializer/json:get_transactions_to_approve - serialize response & deserialize request
close #874

### CODE REVIEW

File: cclient/request/get_transactions_to_approve.c [hunk 1]
```diff
@@ -10,6 +10,7 @@
 get_transactions_to_approve_req_t* get_transactions_to_approve_req_new() {
   get_transactions_to_approve_req_t* req =
       (get_transactions_to_approve_req_t*)calloc(1, sizeof(get_transactions_to_approve_req_t));
+  req->reference = (flex_trit_t*)calloc(1, sizeof(flex_trit_t) * FLEX_TRIT_SIZE_243);
```
Comment (thibault-martinez) [inline cclient/request/get_transactions_to_approve.c]: `reference` is optional, that's why it's not allocated in `get_transactions_to_approve_req_new`.
It is allocated only if needed in `get_transactions_to_approve_req_set_reference`.

**Review by thibault-martinez — CHANGES_REQUESTED** [review]

File: cclient/serialization/json/get_transactions_to_approve.c [hunk 2]
```diff
@@ -50,25 +45,77 @@ retcode_t json_get_transactions_to_approve_serialize_request(const serializer_t
...
   JSON_CHECK_ERROR(json_obj, json_item, json_logger_id);
 
-  ret = json_string_hash_to_flex_trits(json_obj, "trunkTransaction", out->trunk);
+  ret = json_get_uint32(json_obj, "depth", &out->depth);
   if (ret != RC_OK) {
     goto end;
   }
 
-  ret = json_string_hash_to_flex_trits(json_obj, "branchTransaction", out->branch);
+  ret = json_string_hash_to_flex_trits(json_obj, "reference", out->reference);
```
Comment (oopsmonk) [inline cclient/serialization/json/get_transactions_to_approve.c]: `reference` is optional, please check if it is set.

**Review by oopsmonk — CHANGES_REQUESTED** [review]

File: cclient/serialization/json/get_transactions_to_approve.c (hunk 2 above)
Comment (thibault-martinez) [reply cclient/serialization/json/get_transactions_to_approve.c]: @HowJMay Please check with `cJSON_HasObjectItem` before

File: cclient/serialization/json/tests/get_transactions_to_approve.c [hunk 3]
```diff
@@ -7,10 +7,10 @@
 
 #include "cclient/serialization/json/tests/shared.h"
 
-void test_serialize_get_transactions_to_approve(void) {
+void test_get_transactions_to_approve_serialize_request(void) {
```
Comment (thibault-martinez) [inline cclient/serialization/json/tests/get_transactions_to_approve.c:10]: Test with no reference ?

File: cclient/serialization/json/tests/get_transactions_to_approve.c [hunk 4]
```diff
@@ -19,22 +19,70 @@ void test_serialize_get_transactions_to_approve(void) {
...
+
   serializer.vtable.get_transactions_to_approve_serialize_request(&serializer, get_tx_approve, serializer_out);
   TEST_ASSERT_EQUAL_STRING(json_text, serializer_out->data);
 
   char_buffer_free(serializer_out);
   get_transactions_to_approve_req_free(&get_tx_approve);
 }
 
-void test_deserialize_get_transactions_to_approve(void) {
+void test_get_transactions_to_approve_deserialize_request(void) {
```
Comment (thibault-martinez) [inline cclient/serialization/json/tests/get_transactions_to_approve.c:53]: Test with no reference ?

**Review by thibault-martinez — CHANGES_REQUESTED** [review]

**Review by thibault-martinez — APPROVED** [review]

**Review by oopsmonk — APPROVED** [review]

**Pull request merged**